### PR TITLE
feat(UIForms): KeyValue widget

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
*WARNING*: This is only on UIForms, not the current forms
No key/value widget

**What is the chosen solution to this problem?**
* Migrate the current keu/value widget
* Change the format from string `lol=mdr' to object `{ key: 'lol', value: 'mdr' }`
* Add a way to disable reorder on Array to have an array of key/value. Next step would be to change the theme of the array component to have more simple arrays on simple inputs.

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/forms.eslint.txt [hunk 1]
```diff
@@ -13,5 +13,27 @@ The react/require-extension rule is deprecated. Please use the import/extensions
...
+  73:3  error  Prop type `object` is forbidden  react/forbid-prop-types
+
+/home/travis/build/Talend/ui/packages/forms/src/widgets/ListViewWidget/ListViewWidget.js
+  123:3  error  Prop type `any` is forbidden  react/forbid-prop-types
+
+/home/travis/build/Talend/ui/packages/forms/src/widgets/TabsWidget.js
+  56:3  error  Prop type `object` is forbidden  react/forbid-prop-types
+  67:3  error  Prop type `object` is forbidden  react/forbid-prop-types
+  68:3  error  Prop type `object` is forbidden  react/forbid-prop-types
+  69:3  error  Prop type `object` is forbidden  react/forbid-prop-types
```
Comment (jmfrancois) [inline output/forms.eslint.txt]: could remove this lint from our official lint ?
We have so many case where shape is not appropriate..

File: packages/forms/src/UIForm/fields/KeyValue/KeyValue.component.test.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+					},
+				],
+			},
+		};
+
+		// when
+		const wrapper = shallow(<KeyValue {...autoFocusProps} />);
+
+		// then
+		expect(wrapper.node).toMatchSnapshot();
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/KeyValue/KeyValue.component.test.js]: please use wrapper.getNode

File: packages/forms/src/UIForm/fields/KeyValue/KeyValue.component.test.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+				...props.schema,
+				readOnly: true,
+			},
+		};
+
+		// when
+		const wrapper = shallow(<KeyValue {...readOnlyProps} />);
+
+		// then
+		expect(wrapper.node).toMatchSnapshot();
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/KeyValue/KeyValue.component.test.js]: please use wrapper.getNode

File: packages/forms/src/UIForm/fields/KeyValue/KeyValue.component.test.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+		},
+		value: { key: 'lol', value: 'mdr' },
+	};
+
+	it('should render default KeyValue', () => {
+		// when
+		const wrapper = shallow(<KeyValue {...props} />);
+
+		// then
+		expect(wrapper.node).toMatchSnapshot();
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/KeyValue/KeyValue.component.test.js]: please use wrapper.getNode

File: packages/forms/src/UIForm/fieldsets/Array/ArrayItem.component.test.js [hunk 5]
```diff
@@ -43,6 +43,25 @@ describe('Array Item component', () => {
...
+				index={3}
+				onRemove={jest.fn()}
+				value={{ }}
+			>
+				<span>This is the item content</span>
+			</ArrayItem>
+		);
+
+		// then
+		expect(wrapper.node).toMatchSnapshot();
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fieldsets/Array/ArrayItem.component.test.js]: plase use wrapper.getNode()

File: output/forms.eslint.txt (hunk 1 above)
Comment (jsomsanith) [reply output/forms.eslint.txt]: This is weird, we only have 1 eslintrc now at root and we have `"react/forbid-prop-types": [0]`
Run `yarn lint:es` locally, and I have not those issues.
Let's wait for next travis run

Comment (jsomsanith) [reply output/forms.eslint.txt]: The rule is already removed from our tool repo

Comment (jsomsanith) [reply output/forms.eslint.txt]: Hmm ok, seems to have a ci issue. I removed the linter results, the ci should recreate them, but they are still removed ...

File: packages/forms/src/UIForm/fields/KeyValue/KeyValue.component.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+	schema: { type: 'string' },
+	type: 'text',
+};
+
+/**
+ * Get last item in array
+ */
+function getLast(array = []) {
+	return array[array.length - 1];
+}
```
Comment (talend-glorieux) [inline packages/forms/src/UIForm/fields/KeyValue/KeyValue.component.js]: You might be able to use lodash's last method instead.

**Pull request merged**